src/installer/tests/HostActivation.Tests/DotnetArgValidation.cs

```diff
@@ -68,6 +68,55 @@ public void MissingArgumentValue_Fails()
                 .And.HaveStdErrContaining($"Failed to parse supported options or their values:");
         }
 
+        [Fact]
+        public void NonManagedFileWithDirectorySeparator_ShowsSpecificError()
+        {
+            // Create a non-.dll/.exe file with directory separator in path
+            string testFile = Path.Combine(sharedTestState.BaseDirectory.Location, "test.txt");
+            File.WriteAllText(testFile, "test content");
+
+            TestContext.BuiltDotNet.Exec(testFile)
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .Execute()
+                .Should().Fail()
+                .And.HaveStdErrContaining($"The application '{testFile}' is not a managed .dll or .exe.");
+        }
+
+        [Fact]
+        public void CommandNameWithoutDirectorySeparator_RoutesToSDK()
+        {
+            // Create a file named "build" in the current directory to simulate a potential command file
+            string buildFile = Path.Combine(sharedTestState.BaseDirectory.Location, "build");
+            File.WriteAllText(buildFile, "#!/bin/bash\necho 'fake build'");
+
+            // Test that "dotnet build" still routes to SDK, not to the file
+            TestContext.BuiltDotNet.Exec("build")
+                .WorkingDirectory(sharedTestState.BaseDirectory.Location)
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .Execute()
+                .Should().Fail()
+                .And.HaveStdErrContaining("The command could not be loaded, possibly because:"); // This should be the SDK resolver error, not our specific error
+        }
+
+        [Fact]
+        public void RelativePathToNonManagedFile_ShowsSpecificError()
+        {
+            // Test relative path with directory separator
+            Directory.CreateDirectory(Path.Combine(sharedTestState.BaseDirectory.Location, "subdir"));
+            string testFile = Path.Combine(sharedTestState.BaseDirectory.Location, "subdir", "test.json");
+            File.WriteAllText(testFile, "{}");
+
+            TestContext.BuiltDotNet.Exec("./subdir/test.json")
+                .WorkingDirectory(sharedTestState.BaseDirectory.Location)
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .Execute()
+                .Should().Fail()
+                .And.HaveStdErrContaining("is not a managed .dll or .exe.");
```

Comment:
This should check for the entire string of "The application '<app>' is not a managed .dll or .exe", where <app> is the argument passed in on the command line.